packages/common/driver-definitions/src/storage.ts

```diff
@@ -385,6 +385,32 @@ export interface IDocumentServiceFactory {
 	): Promise<IDocumentService>;
 }
 
+export interface ISocketStorageDiscoveryParts {
+	/**
+	 * Niche interface to allow odsp-driver to expose specific pieces necessary for /broadcast-signal
+	 * endpoint. Please do not use this for anything else. We will phase this out once we know how
+	 * the interface for a more uniform interface across multiple drivers looks.
+	 */
+
+	// SPO gives us runtimeTenantId, we remap it to tenantId
+	// See getSocketStorageDiscovery
+	runtimeTenantId?: string;
+	tenantId: string;
+
+	/**
+	 * PUSH URL
+	 */
+	deltaStreamSocketUrl: string;
+}
+
+export interface IProvideSessionAwareDriverFactory {
+	readonly ISessionAwareDriverFactory: ISessionAwareDriverFactory;
+}
+
+export interface ISessionAwareDriverFactory extends IProvideSessionAwareDriverFactory {
```

Comment:
Trello component is going to need to use this as described in the Usage section in the PR description. The Host (Loop) will pass in the documentServiceFactory and the resolvedUrl using providers and then Trello component will need to use FluidObject to interrogate the driver factory instance to see if it can call call the getRelaySessionService function.